cmd/dagger-graph/vertex.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	bkclient "github.com/moby/buildkit/client"
+)
+
+type WrappedVertex struct {
+	v *bkclient.Vertex
+}
+
+func (w WrappedVertex) ID() string {
+	return w.v.Digest.String()
+}
+
+func (w WrappedVertex) FullName() string {
+	path := []string{}
+	for _, p := range w.Pipeline() {
+		path = append(path, p.Name)
+	}
+	path = append(path, fmt.Sprintf("%q", w.ID()))
+	return strings.Join(path, ".")
+}
+
+func (w WrappedVertex) Name() string {
+	var custom CustomName
+	if json.Unmarshal([]byte(w.v.Name), &custom) == nil {
+		return custom.Name
+	}
+	return w.v.Name
+}
+
+func (w WrappedVertex) Pipeline() PipelinePath {
+	var custom CustomName
+	if json.Unmarshal([]byte(w.v.Name), &custom) == nil {
+		if len(custom.Pipeline) > 0 {
+			return custom.Pipeline
+		}
+	}
+
+	pg := w.v.ProgressGroup.GetId()
+	if pg != "" {
+		var pipeline PipelinePath
+		if json.Unmarshal([]byte(pg), &pipeline) == nil {
+			return pipeline
+		}
+	}
+	return PipelinePath{}
+}
+
+func (w WrappedVertex) Internal() bool {
+	var custom CustomName
+	if json.Unmarshal([]byte(w.v.Name), &custom) == nil {
+		if custom.Internal {
+			return true
+		}
+	}
+	return false
+}
+
+func (w WrappedVertex) Inputs() []string {
+	inputs := make([]string, 0, len(w.v.Inputs))
+	for _, i := range w.v.Inputs {
+		inputs = append(inputs, i.String())
+	}
+	return inputs
+}
+
+type Pipeline struct {
+	Name        string
+	Description string
+}
+
+type PipelinePath []Pipeline
+
+type CustomName struct {
+	Name     string
+	Internal bool
+	Pipeline PipelinePath
+}
```

Comment:
Can we use the actual structs in `core/pipeline.go` now that the PR for them has been merged?

Reply:
Done!